### ISSUE TITLE
bug: `ion-toggle` and `ion-checkbox` have no focus indicators

### ISSUE THREAD

**Issue by darrylnoakes** [ionic-team/ionic-framework#25884]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

`ion-toggle`s and `ion-checkbox`es have no focus indicators themselves. The wrapping `ion-item` provides focus indicators, which is not always available.

### Expected Behavior

`ion-toggle`s and `ion-checkbox`es should have a way to enable focus indicators on themselves when navigated to with the keyboard.

### Steps to Reproduce

Simply use an `ion-toggle` and/or `ion-checkbox` and tab to it.
The demos in the docs are sufficient.

### Ionic Info

```
Ionic:

   Ionic CLI       : 6.16.3 (C:\Users\Darryl\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework : @ionic/vue 6.2.5

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v18.4.0 (C:\Program Files\nodejs\node.exe)
   npm    : 8.12.1
   OS     : Windows 10
```

### Additional Information

I am using a temporary fix for this.

For an `ion-item` with a label and a checkbox:

```scss
/*
 * Focus outline SCSS variables
 */
$focusOutlineWidth: 2px;
$focusOutlineOffset: 2px;
$focusOutlineSize: $focusOutlineWidth + $focusOutlineOffset;

/*
 * Adjustments for focus outline
 */
.item-with-checkbox {
  .md & ion-label {
    margin-left: 8px;
  }

  // Adjust padding to account for focus outline.
  --padding-start: calc(var(--padding-item) - #{$focusOutlineSize});

  & ion-checkbox {
    // Focus state styles

    border-radius: var(--border-radius);
    margin-left: $focusOutlineSize;
    outline-offset: $focusOutlineOffset;

    &.focused {
      outline: $focusOutlineWidth solid var(--ion-color-primary);
    }
  }
}
```

Result:

MD:
![MD result](https://user-images.githubusercontent.com/67335139/188650612-057d773a-7f1e-4fe9-987d-3d4ed7016a9a.png)

iOS:
![iOS result](https://user-images.githubusercontent.com/67335139/188650700-fe1ea527-1564-4faa-bed6-4d519cbad13b.png)

**Comment by liamdebeasi** [comment]
Thanks for the report. Changes to toggle + checkbox focus indicators are being tracked in https://github.com/ionic-team/ionic-framework/issues/18279.
